### PR TITLE
Change size and diskimage flags as "not required"

### PR DESCRIPTION
**Pull request by haardikdharma10**
### Background

https://github.com/civo/cli/issues/200

### CODE REVIEW

File: cmd/instance_create.go [hunk 1]
```diff
@@ -73,12 +52,12 @@ If you wish to use a custom format, the available fields are:
 			os.Exit(1)
 		}
 
-		if hostnameCreate != "" {
-			if utility.ValidNameLength(hostnameCreate) {
+		if len(args) > 0 {
```
Comment (alejandrojnm) [inline cmd/instance_create.go:84]: I think we can change this to use https://github.com/civo/cli/blob/fa542ad26368fe8fe797f4057912a1e655525b82/utility/random.go#L44 in the case of the user don't pass the hostname

File: cmd/instance_create.go [hunk 2]
```diff
@@ -16,34 +16,13 @@ import (
...
-	* ssh_key
-	* notes
-	* firewall_id
-	* tags
-	* script
-	* created_at
-	* reverse_dns
-	* private_ip
-	* public_ip`,
+	Example: "civo instance create HOSTNAME [flags]",
```
Comment (alejandrojnm) [inline cmd/instance_create.go:25]: Why we remove the available fields??

File: cmd/instance_create.go (hunk 1 above)
Comment (haardikdharma10) [reply cmd/instance_create.go:84]: @alejandrojnm if the user doesn't pass the hostname it will be automatically generated from - https://github.com/civo/civogo/blob/master/instance.go#L176 

it is tested and verified 👍

File: cmd/instance_create.go (hunk 2 above)
Comment (haardikdharma10) [reply cmd/instance_create.go:25]: Added

**Comment by andyjeffries** [comment]
Hold on, this doesn't implement the bug fix required. Specifically this is removing the flag and changing it to be the correct/expected behaviour. However, people may have scripts that rely on the flag - so BOTH mechanisms need to be supported.

**Comment by haardikdharma10** [comment]
I am not removing the flags for size and diskimages. I am just marking them as "not required". Did I misunderstood something?

**Comment by andyjeffries** [comment]
This comment only relates to this merge request. This MR removes the hostname flag. It shouldn't remove it but accept EITHER the hostname flag or the first positional parameter. These should be equivalent:

```
civo instance create my-demo
civo instance create -s my-demo
civo instance create --hostname=my-demo
```

**Comment by andyjeffries** [comment]
(the stuff about size, disk images is in a separate MR - or missing as it stands)

**Comment by andyjeffries** [comment]
Ohhh, I see the title of the MR is one thing, but in the code you do things with disk image and size too. OK, I'll address those:

You seem to have removed the "required" part of the CLI spec. However, if you don't specify a disk image it will error as the code stands, correct? That's not right behaviour. It should have a default disk image in there.

In short, this command should work - can you test it your side please:

```
civo instance create my-demo
```

No specification of size (medium is OK) nor disk image (Ubuntu 20.04 should be default - for now)

In fact, even this should work:

```
civo instance create
```

(it should generate a random hostname, I think it already does this though)

**Comment by haardikdharma10** [comment]
Okay, so I should have changed the PR title. This MR indeed removes the hostname flag and adds it as an argument. This needs to be changed as you mentioned. The stuff about size and disk images is also in this MR. The error when creating instance with `civo instance create dokku-demo` was `Error: required flag(s) "diskimage", "size" not set` 

This was because they were marked as required flags.

Ahh I see your recent comment, all the commands that you have mentioned have been tested by me :) 

To summarize - 

- `civo instance create my-demo` creates an instance with hostname `my-demo` size `g3.medium` and disk image `Ubuntu 20.04`
-  `civo instance create` creates an instance with a random hostname size `g3.medium` and disk image `Ubuntu 20.04`
- `civo instance create new-instance -t ubuntu-focal -i g3.medium` creates an instance with hostname `new-instance` and the mentioned size and disk image.

**Comment by andyjeffries** [comment]
Cool, thank you - the last part is key for me.

**Comment by andyjeffries** [comment]
I am a bit lost though - where is Ubuntu 20.04 or focal specified as the default value? I can't find it on https://github.com/civo/cli/pull/203/files and the API code doesn't seem to have it (don't link to the API code here, send that in Slack if it's in there).

**Comment by haardikdharma10** [comment]
@andyjeffries it is here - https://github.com/civo/civogo/blob/master/instance.go#L169 

Wondering should it be bionic or focal?

**Comment by andyjeffries** [comment]
Should be Focal for now, the latest LTS version of Ubuntu (until Jammy is released - 22.4)

**Comment by andyjeffries** [comment]
Thanks, had forgotten about civogo being in the mix.

**Review by alejandrojnm — APPROVED** [review]
LGTM
